src/libotutil/ot-checksum-utils.h

```diff
@@ -21,19 +21,44 @@
 
 #pragma once
 
-#include <gio/gio.h>
+#include "libglnx.h"
 
 G_BEGIN_DECLS
 
 void ot_bin2hex (char *out_buf, const guint8 *inbuf, gsize len);
 
 guchar *ot_csum_from_gchecksum (GChecksum *checksum);
 
+struct OtChecksum {
+  gboolean initialized;
+  guint uints[2];
+  gpointer data[2];
+};
+typedef struct OtChecksum OtChecksum;
+
+/* Same as OSTREE_SHA256_DIGEST_LEN, but this header can't depend on that */
```

Comment:
I think I missed the part about why we can't do that. Can you expand on this comment?

Reply:
We could, but it feels like a layering violation since conceptually `src/libostree` depends on `src/libotutil` (which depends on `libglnx/`), etc.

Reply:
I came up with something that'll at least error out if we somehow make them different :arrow_dow